uwblueprint/planet-read
# Add translator_last_activity and reviewer_last_activity time columns…

… to StoryTranslation model

## Notion ticket link



[Add translator_last_activity and reviewer_last_activity time columns to StoryTranslation model](https://www.notion.so/uwblueprintexecs/Add-translator_last_activity-and-reviewer_last_activity-time-columns-to-StoryTranslation-model-35720c1545224b138eaf3d0c6cf35ade)



## Implementation description

- Added time columns to database
- Created new update activity mutation that is called after:
    - storytranslationcontent edits
    - storytranslationcontent status changes
    - comments made
    - comments resolved



## Steps to test

1. Check that the columns `reviewer_last_activity` and `translator_last_activity` have been added to the `story_translations` and `story_translations_all` tables using mysql
2. The `reviewer_last_activity` and `translator_last_activity` columns should have null entries
3. Login as `planetread+carlsagan@uwblueprint.org`
4. Edit Great Gatsby translation
5. View the `story_translations` table again, the appropriate timestamp (row: id 10, column: translator_last_activity) should be updated (note that the time zone is utc)
6. Make a comment on Great Gatsby translation
7. View the `story_translations` table again, the appropriate timestamp (row: id 10, column: translator_last_activity) should be updated (note that the time zone is utc)
8. Send Great Gatsby translation for review
9. Sign out
10. Login as `planetread+dwightdeisenhower@uwblueprint.org`
11. Update status on Great Gatsby translation
12. View the `story_translations` table again, the appropriate timestamp (row: id 10, column: reviewer_last_activity) should be updated (note that the time zone is utc)
13. Make a comment on Great Gatsby translation
14. View the `story_translations` table again, the appropriate timestamp (row: id 10, column: reviewer_last_activity) should be updated (note that the time zone is utc)
15. You can repeat these steps with other users and translations as well



## What should reviewers focus on?

- Does the time zone make sense? (I am using utc to align with the comment timestamps)
- Resolving comments seems to be WIP, so I didn't test it (please let me know if it should be working)
- Does the implementation make sense? I created a `UpdateStoryTranslationLastActivity` mutation which is called after every applicable mutation (`UPDATE_STORY_TRANSLATION_CONTENTS`, `UPDATE_STORY_TRANSLATION_CONTENT_STATUS`, `CREATE_COMMMENT`, `UPDATE_COMMENT_BY_ID`)

## Checklist
